### PR TITLE
Audited file system tests.

### PR DESCRIPTION
**Pull request by AMZN-scspaldi**
Signed-off-by: scspaldi <scspaldi@amazon.com>

## What does this PR do?

This is an audit of the file system tests. Redundant/pointless tests were removed and new edge cases were covered.

## How was this PR tested?

Ran locally. All unit tests passed.

### CODE REVIEW

**Review by evanchia-ly-sdets — APPROVED** [review]

File: Tools/LyTestTools/tests/unit/test_file_system.py [hunk 1]
```diff
@@ -59,6 +59,11 @@ def test_CheckFreeSpace_EnoughSpace_NoRaise(self, mock_disk_usage):
 
 class TestSafeMakedirs(unittest.TestCase):
 
+    @mock.patch('os.makedirs')
+    def test_SafeMakedirs_ValidCreation_Success(self, mock_makedirs):
+        file_system.safe_makedirs('dummy')
```
Comment (Kadino) [inline Tools/LyTestTools/tests/unit/test_file_system.py:64]: This function hides most of its side effects, so this is a reasonable sanity test despite there being no test assertion. Though I notice the function's docstring/comments lies about how it emits errors.

**Review by Kadino — APPROVED** [review]
New tests look fine. I'm okay with removing these heavily mocked tests of utilities in a file which virtually no test uses (due to involving destructive edits of workspace-automation, which is more suited for personal workspace automation).
